client/ayon_core/hosts/max/lib.py

```diff
@@ -0,0 +1,38 @@
+import os
+from ayon_core.settings import get_project_settings
+from ayon_core.lib import Logger
+
+
+def create_workspace_mxp(workdir, project_name, project_settings=None):
+    dst_filepath = os.path.join(workdir, "workspace.mxp")
+    if os.path.exists(dst_filepath):
+        return
+
+    if not os.path.exists(workdir):
+        os.makedirs(workdir)
+
+    if not project_settings:
+        project_settings = get_project_settings(project_name)
+    log = Logger.get_logger("create_workspace_mxp")
+    mxp_workspace = project_settings["max"].get("mxp_workspace")
+    # Ensure the hook would not cause possible error
+    # when using the old addon.
+    if mxp_workspace is None:
+        log.debug("No mxp workspace setting found in the "
+                  "latest Max Addon. Please update to 0.1.8")
+        return
+    if mxp_workspace.get("enabled_project_creation"):
+        max_script = mxp_workspace.get("mxp_workspace_script")
+        # Skip if mxp script in settings is empty
+        if not max_script:
+            log.debug("File 'workspace.mxp' not created. Settings value is empty.")
+            return
+
+        edited_script = "\n".join((
+            '[Directories]',
+            f'ProjectFolder={workdir}'
+        ))
+        max_script = max_script.replace("[Directories]", edited_script)
```

Comment:
What if `mxp_workspace_script` in setting has no `[Directories]` string set? Is that totally fine?

Also, what if `ProjectFolder=` value is already defined in settings as well? Is it fine if it's defined more than once in the `.mxp` file?

Reply:
we can use rt.pathconfig.setCurrentProjectFolder(workdir) instead of adding this.